src/chainlock/chainlock.cpp

```diff
@@ -490,4 +491,72 @@ void CChainLocksHandler::Cleanup()
         }
     }
 }
+
+void CChainLocksHandler::QueueCoinbaseChainLock(const chainlock::ChainLockSig& clsig)
+{
+    LOCK(cs);
+
+    if (!IsEnabled()) {
+        return;
+    }
+
+    // Only queue if it's potentially newer than what we have
+    if (!bestChainLock.IsNull() && clsig.getHeight() <= bestChainLock.getHeight()) {
+        return;
+    }
+
+    // Check if we've already seen this chainlock
+    const uint256 hash = ::SerializeHash(clsig);
+    if (seenChainLocks.count(hash) != 0) {
+        return;
+    }
+
+    pendingCoinbaseChainLocks.push_back(clsig);
```

Comment:
1. is it true, that _every_ chainlock_ from coinbase will be added to `pendingCoinbaseChainLocks`?